change/@ni-nimble-components-184413cc-afcf-4db1-b373-1bdb1a98f015.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "updates the list button to be higlighted only on last node type",
+  "packageName": "@ni/nimble-components",
+  "email": "123591928+saikrishnan-ni@users.noreply.github.com",
+  "dependentChangeType": "patch"
+}
```

Comment:
I was able to get into a state testing this where the rendered markdown has different types of bullets at the same indentation level:

![image](https://github.com/ni/nimble/assets/1588923/6c72788c-3cd2-4329-adb5-b21c1a22881a)
```
Plain text

* sdf

  1. sdf

  2. 

  * asd

  * sfd

  1. 
```

![mixedlist](https://github.com/ni/nimble/assets/1588923/43e659e0-639c-4045-9989-dd1d59519cc8)

The above gif reproduces how to get in that state:
1. create a list
2. create a sublist of a different type
3. Add an additional item to the sublist, but decide it should be top-level with shift+tab
4. Type something but realize it should actually be in the sublist so press tab. Notice that the parent list type gets pulled to the sublist type. Now the sublist has two different types.

Maybe a bug should be created for that behavior? It seems unrelated to this change. We should either find a way to prevent mixing sublist types like that or have tests for it. I also don't know how the viewer should behave is it recieves markdown with a mixed sublist type.

Reply:
@rajsite 
We're able to get this rendered properly in the viewer like below.
![image](https://github.com/ni/nimble/assets/123591928/ff25ad17-d418-4153-a3ce-6db9bfd52ac0)

And yes, this behavior is reproduceable. Unfortunately, this looks like a behavior from Tiptap and we're able to see this behavior in Azdo and Github as well as below.
![image](https://github.com/ni/nimble/assets/123591928/237813e7-a6ed-4723-8d0c-c121f412e28a)
![image](https://github.com/ni/nimble/assets/123591928/6be16ade-c8bc-4018-b5bf-8260576637de)

Please do post your thoughts in the group on this and we can take up this discussion in a new thread or issue.

Reply:
We should either find a way to prevent mixing sublist types like that or have tests for it. Sounds like it might be expected behavior so we was should have tests for it. Please capture as a tech debt issue and have it addressed in a follow up pr

Reply:
Created a new TD https://github.com/ni/nimble/issues/1473 for follow up PR and resolving this conversation.